examples_tests/50.MeshPacking/main.cpp

```diff
@@ -0,0 +1,388 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#define _NBL_STATIC_LIB_
+#include <iostream>
+#include <cstdio>
+#include <nabla.h>
+
+//! I advise to check out this file, its a basic input handler
+#include "../common/QToQuitEventReceiver.h"
+#include "nbl/asset/CCPUMeshPackerV1.h"
+#include "nbl/asset/CCPUMeshPackerV2.h"
+
+using namespace nbl;
+using namespace core;
+using namespace asset;
+using namespace video;
+
+constexpr const char* SHADER_OVERRIDES[2] =
+{
+R"(
+#define _NBL_VERT_INPUTS_DEFINED_
+#define _NBL_VERT_MAIN_DEFINED_
+
+//pos
+layout(set = 0, binding = 0) uniform samplerBuffer MeshPackedData_R32G32B32_SFLOAT;
+
+//uv
+//layout(set = 0, binding = 1) uniform samplerBuffer MeshPackedData_R32G32_SFLOAT;
+
+//normal
+layout(set = 0, binding = 2) uniform samplerBuffer MeshPackedData_A2B10G10R10_SNORM_PACK32;
+
+layout(set = 0, binding = 3) readonly buffer VertexDataOffsetTable
+{
+    uint dataOffsetTable[];
+} vertexPosition;
+
+layout(set = 0, binding = 4) readonly buffer VertexNormalOffsetTable
+{
+    uint dataOffsetTable[];
+} vertexNormal;
```

Comment:
this sort of stuff should come from a builtin GLSL header (or header generator) provided by the packer

Reply:
user should have ubo/ssbo/tbo choice though, right?

Reply:
yes

UBO is pointless because their minimum size is 64kb and only AMD allows much more than that, also UBOs are not optimized towards non-uniform data loads